Source/WebKit/WebProcess/WebPage/WebFrame.cpp

```diff
@@ -415,12 +415,17 @@ void WebFrame::createProvisionalFrame(ProvisionalFrameCreationParameters&& param
     m_provisionalFrame = localFrame.ptr();
     localFrame->init();
 
+    remoteFrame->setIsProvisional(true);
+    localFrame->setIsProvisional(true);
+
     if (parameters.layerHostingContextIdentifier)
         setLayerHostingContextIdentifier(*parameters.layerHostingContextIdentifier);
 }
 
 void WebFrame::destroyProvisionalFrame()
 {
+    if (auto* remoteFrame = coreRemoteFrame())
+        remoteFrame->setIsProvisional(false);
```

Comment:
It's the LocalFrame we need to set as non-provisional here, even though it will likely be destroyed but there may still be a strong reference to it from something else.